cpp/include/raft/neighbors/nn_descent_types.hpp

```diff
@@ -85,6 +87,7 @@ struct index : ann::index {
       res_{res},
       metric_{raft::distance::DistanceType::L2Expanded},
       graph_{raft::make_host_matrix<IdxT, int64_t, row_major>(n_rows, n_cols)},
+      distances_{raft::make_host_matrix<DistData_t, int64_t, row_major>(n_rows, n_cols)},
```

Comment:
This is an eager allocation. Instead, the allocation should only occur if `return_distances=true`